### PR TITLE
Add Car controller with providing Car entity 

### PR DESCRIPTION
**Pull request by Recdt**
* Added car entity
* Added Liquibase script for cars table
* Added DTOs
* Added mapper
* Added service layer for car
* Added car repository

### CODE REVIEW

File: src/main/java/online/carsharing/dto/request/CarRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
+import java.math.BigDecimal;
+import lombok.Data;
+import online.carsharing.entity.Car;
+
+@Data
+public class CarRequestDto {
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/dto/request/CarRequestDto.java]:
```suggestion
public class CreateCarRequestDto {
```

File: src/main/java/online/carsharing/dto/request/CarRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @NotBlank
+    private String model;
+    @NotBlank
+    private String brand;
+    @PositiveOrZero
+    @NotNull
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/dto/request/CarRequestDto.java]:
```suggestion
```

File: src/main/java/online/carsharing/dto/update/CarUpdateDto.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package online.carsharing.dto.update;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
+import lombok.Data;
+
+@Data
+public class CarUpdateDto {
+    @PositiveOrZero
+    @NotNull
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/dto/update/CarUpdateDto.java]:
```suggestion
```

File: src/main/java/online/carsharing/repository/car/CarRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package online.carsharing.repository.car;
+
+import online.carsharing.entity.Car;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/repository/car/CarRepository.java]:
```suggestion
```
not necessary

File: src/main/java/online/carsharing/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        return carRepository.findAll(pageable).stream()
+                .map(carMapper::toDto)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public CarResponseDto updateById(Long carId, CarUpdateDto carUpdateDto) {
+        checkExistence(carId);
+        Car car = carRepository.getReferenceById(carId);
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/service/impl/CarServiceImpl.java]:
```suggestion
        Car car = carRepository.findById(carId);
```
I quess we should fetch all info about car here from db. When we use getReferenceById, we actually fetch only Id

File: src/main/java/online/carsharing/service/impl/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        car.setInventory(carUpdateDto.getInventory());
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Transactional
+    @Override
+    public CarResponseDto getById(Long carId) {
+        checkExistence(carId);
+        return carMapper.toDto(carRepository.getReferenceById(carId));
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/service/impl/CarServiceImpl.java]:
```suggestion
        return carMapper.toDto(carRepository.findById(carId));
```

File: src/main/resources/db/changelog/changes/create-cars-table.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        - createTable:
+            tableName: cars
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (HAIOVYI) [inline src/main/resources/db/changelog/changes/create-cars-table.yml]:
```suggestion
```

File: src/main/resources/db/changelog/changes/create-cars-table.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+              - column:
+                  name: brand
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: type
+                  type: enum('SEDAN', 'SUV', 'HATCHBACK', 'UNIVERSAL')
+                  constraints:
+                    nullable: false
```
Comment (HAIOVYI) [inline src/main/resources/db/changelog/changes/create-cars-table.yml]: so how we doo???? with varchar(255) or with enums?? )

File: src/main/resources/db/changelog/changes/create-cars-table.yml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                  type: decimal(19, 2)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
+
```
Comment (HAIOVYI) [inline src/main/resources/db/changelog/changes/create-cars-table.yml]: I don't understand exactly, but here we have some issues with emty space

**Review by HAIOVYI — CHANGES_REQUESTED** [review]
Good job, but I think we need to communicate in our meetings some general issues

File: src/main/java/online/carsharing/service/impl/CarServiceImpl.java (hunk 6 above)
Comment (Recdt) [reply src/main/java/online/carsharing/service/impl/CarServiceImpl.java:49]: findById deprecated

File: src/main/resources/db/changelog/changes/create-cars-table.yml (hunk 8 above)
Comment (azzilian) [reply src/main/resources/db/changelog/changes/create-cars-table.yml]: I think varchar is better to go we do not need any restrictions in demands

File: src/main/java/online/carsharing/entity/Car.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    private BigDecimal dailyFee;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+    public enum Type {
+        SEDAN,
+        SUV,
+        HATCHBACK,
+        UNIVERSAL
+    }
```
Comment (azzilian) [inline src/main/java/online/carsharing/entity/Car.java]: maybe better to create in separate file?

**Review by azzilian — CHANGES_REQUESTED** [review]
I think Type should be separated into CarType(or something like this) enum

File: src/main/java/online/carsharing/entity/Car.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    @Column(nullable = false)
+    private String brand;
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private Type type;
+    @Column(nullable = false)
+    private int inventory;
+    @Column(nullable = false)
+    private BigDecimal dailyFee;
+}
```
Comment (nklimovych) [inline src/main/java/online/carsharing/entity/Car.java]: Add empty lines between variables
```suggestion
public class Car {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String model;
    
    @Column(nullable = false)
    private String brand;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Type type;
    
    @Column(nullable = false)
    private int inventory;
    
    @Column(nullable = false)
    private BigDecimal dailyFee;
}
```

File: src/main/resources/db/changelog/changes/chopyk/create-cars-table.yaml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: daily_fee
+                  type: decimal(19, 2)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: deleted
+                  type: boolean
```
Comment (nklimovych) [inline src/main/resources/db/changelog/changes/chopyk/create-cars-table.yaml]:
```suggestion
                  type: bit
```

**Review by nklimovych — APPROVED** [review]

**Review by azzilian — APPROVED** [review]
Gj

**Review by HAIOVYI — APPROVED** [review]

**Pull request merged**